File: M8/Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/shipay/requests/CreateOrderRequest.java
package com.elgin.elginexperience.shipay.requests;

import com.elgin.elginexperience.shipay.models.Buyer;
import com.elgin.elginexperience.shipay.models.OrderItem;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class CreateOrderRequest {
    public CreateOrderRequest(String order_ref, String wallet, float total, List<OrderItem> items, Buyer buyer) {
        this.order_ref = order_ref;
        this.wallet = wallet;
        this.total = total;
        this.items = items;
        this.buyer = buyer;
    }
    @SerializedName("order_ref")
    public String order_ref;

    @SerializedName("wallet")
    public String wallet;

    @SerializedName("total")
    public float total;

    @SerializedName("items")
    public List<OrderItem> items;

    @SerializedName("buyer")
    public Buyer buyer;
}

File: Exemplos/App_eXperience_Kotlin/Elgin_AppExperience_Kotlin-font/app/src/main/java/com/example/e1_kotlin_r/shipay/requests/CreateOrderRequest.kt
package com.example.e1_elgin_kotlin.shipay.requests

import com.example.e1_elgin_kotlin.shipay.models.Buyer
import com.example.e1_elgin_kotlin.shipay.models.OrderItem
import com.google.gson.annotations.SerializedName

data class CreateOrderRequest(
    @field:SerializedName("order_ref") var order_ref: String,
    @field:SerializedName("wallet") var wallet: String,
    @field:SerializedName("total") var total: Float,
    @field:SerializedName("items") var items: List<OrderItem>,
    @field:SerializedName("buyer") var buyer: Buyer
)
